code/modules/clothing/suits/miscellaneous.dm

```diff
@@ -438,6 +438,48 @@
 	body_parts_covered = UPPER_TORSO|LOWER_TORSO|ARMS
 	cold_protection = UPPER_TORSO|LOWER_TORSO|ARMS
 
+	verb/toggle()
+		set name = "Toggle Labcoat Buttons"
+		set category = "Object"
+		set src in usr
+
+		if(!usr.canmove || usr.stat || usr.restrained())
+			return 0
+
+		switch(icon_state)
+			if("bomber_open")
+				src.icon_state = "bomber"
+				usr << "You zip up [src]."
+			if("bomber")
+				src.icon_state = "bomber_open"
+				usr << "You unzip [src]."
+			if("bombersec_open")
+				src.icon_state = "bombersec"
+				usr << "You zip up [src]."
+			if("bombersec")
+				src.icon_state = "bombersec_open"
+				usr << "You unzip [src]."
+			else
+				usr << "You attempt to button-up the velcro on your [src], before promptly realising how retarded you are."
+				return
+		usr.update_inv_wear_suit()	//so our overlays update

```

Comment:
This entire proc seems a bit snowflakey---most others go purely by based icon state with some modifier and setting it back to initial---I'm pretty sure that's how jumpsuits work.